test/capi/testPaint.cpp

```diff
@@ -0,0 +1,177 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <thorvg_capi.h>
+#include "catch.hpp"
+
+TEST_CASE("Paint Transform", "[capiPaintTransform]")
+{
+    Tvg_Paint *paint = NULL;
+    Tvg_Matrix matrix_set = {1, 0, 0, 0, 1, 0, 0, 0, 1}, matrix_get;
+
+    paint = tvg_shape_new();
+    REQUIRE(paint != NULL);
+
+    REQUIRE(tvg_paint_transform(paint, &matrix_set) == TVG_RESULT_SUCCESS);
+    REQUIRE(tvg_paint_get_transform(paint, &matrix_get) == TVG_RESULT_SUCCESS);
+    REQUIRE(matrix_get.e11 == matrix_set.e11);
+    REQUIRE(matrix_get.e12 == matrix_set.e12);
+    REQUIRE(matrix_get.e13 == matrix_set.e13);
+    REQUIRE(matrix_get.e21 == matrix_set.e21);
+    REQUIRE(matrix_get.e22 == matrix_set.e22);
+    REQUIRE(matrix_get.e23 == matrix_set.e23);
+    REQUIRE(matrix_get.e31 == matrix_set.e31);
+    REQUIRE(matrix_get.e32 == matrix_set.e32);
+    REQUIRE(matrix_get.e33 == matrix_set.e33);
+
+    REQUIRE(tvg_paint_del(paint) == TVG_RESULT_SUCCESS);
+}
+
+TEST_CASE("Paint Translate", "[capiPaintTranslate]")
+{
+    Tvg_Paint *paint = NULL;
+    Tvg_Matrix matrix_get;
+    float tx = 20, ty = 30;
+
+    paint = tvg_shape_new();
+    REQUIRE(paint != NULL);
+
+    REQUIRE(tvg_paint_translate(paint, tx, ty) == TVG_RESULT_SUCCESS);
+    REQUIRE(tvg_paint_get_transform(paint, &matrix_get) == TVG_RESULT_SUCCESS);
+    REQUIRE(matrix_get.e11 == 1.0f);
+    REQUIRE(matrix_get.e12 == 0.0f);
+    REQUIRE(matrix_get.e13 == tx);
+    REQUIRE(matrix_get.e21 == 0.0f);
+    REQUIRE(matrix_get.e22 == 1.0f);
+    REQUIRE(matrix_get.e23 == ty);
+    REQUIRE(matrix_get.e31 == 0.0f);
+    REQUIRE(matrix_get.e32 == 0.0f);
+    REQUIRE(matrix_get.e33 == 1.0f);
+
+    REQUIRE(tvg_paint_del(paint) == TVG_RESULT_SUCCESS);
+}
+
+TEST_CASE("Paint Scale", "[capiPaintScale]")
+{
+    Tvg_Paint *paint = NULL;
+    Tvg_Matrix matrix_get;
+    float scale = 2.5f;
+
+    paint = tvg_shape_new();
+    REQUIRE(paint != NULL);
+
+    REQUIRE(tvg_paint_scale(paint, scale) == TVG_RESULT_SUCCESS);
+    REQUIRE(tvg_paint_get_transform(paint, &matrix_get) == TVG_RESULT_SUCCESS);
+    REQUIRE(matrix_get.e11 == scale);
```

Comment:
I'm against comparing floats like ints. 
I know we do that sometimes in tvg, but it's because of performance reasons and even if the comparison won't work as we expect, the result still will be ok. and if the comparison won't work here, the test will not pass.
So I'd use fabs - as you did with a rotation sect

Reply:
I'd change that everywhere you have floats

Reply:
I am okay with direct comparison of floats there, for scale and translation, as these are directly set (without add or multiply operations) in `RenderTransform::update`.
Nevertheless I understand your point of view, I will change it to abs and eps comparison.